### PR TITLE
EmuASL rewritten in order to return the address of WRAM

### PR DESCRIPTION
**Pull request by Jujstme**
Previously it was returning a pointer to it, but memory inspection of various emulator led to the conclusion that easily finding a pointer path in memory is not that easy. Instead of having to allocate a new memory page and writing the address to it (which is a hacky solution), it's probably best to just return the baseaddress of WRAM in the first place.

This means:

1. The Loaded directive needs to check whether the WRAM address changed (for emulators that have that behaviour)
2. No longer needing to use DeepPointer in the ASL, eg:

image

The code for the already included emulators has been updated accordingly.

### CODE REVIEW

File: emu/EmuASL/Emulators/Xebra.cs [hunk 1]
```diff
@@ -21,7 +20,7 @@ public Xebra()
 
 		protected override IntPtr Internal_GetWRAM()
 		{
-			return Data.EmuProcess.ReadPointer(_scanner.Scan(_target) + 0x16A);
+			return (IntPtr)Data.EmuProcess.ReadValue<int>(_scanner.Scan(_target) + 0x16A);
```
Comment (just-ero) [inline emu/EmuASL/Emulators/Xebra.cs:23]: This looks suspicious. Why are you doing it like this?

File: emu/EmuASL/Emulators/pSX.cs [hunk 2]
```diff
@@ -6,22 +6,14 @@ namespace EmuASL.Emulators
 {
 	public class pSX : EmulatorBase
 	{
-		public override bool Loaded => Data.Running;
+		int pageCount = Data.EmuProcess.MemoryPages().Count(p => (int)p.RegionSize == 0x201000 && p.State == MemPageState.MEM_COMMIT && p.Type == MemPageType.MEM_PRIVATE);
```
Comment (just-ero) [inline emu/EmuASL/Emulators/pSX.cs:9]: Don't forget the access modifier (`private` in this case). Also, would it make more sense to set this in the constructor?

Comment (just-ero) [reply emu/EmuASL/Emulators/pSX.cs:9]: I find it's less customary to set them the way you are, functionally there shouldn't be a difference.

File: emu/EmuASL/Emulators/Xebra.cs (hunk 1 above)
Comment (Jujstme) [reply emu/EmuASL/Emulators/Xebra.cs:23]: I wasn't sure and was too lazy to check. Yeah, ReadPointer is better.

**Pull request merged**